File: app/src/main/java/com/test/kabak/openweather/core/network/ForecastResponse.java
package com.test.kabak.openweather.core.network;

import com.google.gson.annotations.SerializedName;

import java.util.ArrayList;

public class ForecastResponse {
    @SerializedName("list")
    public ArrayList<ForecastObject> forecasts;

    public static class ForecastObject {
        @SerializedName("dt")
        public long dateTime;

        @SerializedName("main")
        public MainObject main;

        @SerializedName("weather")
        public ArrayList<WeatherObject> weather;

        @SerializedName("wind")
        public WindObject wind;
    }

    public static class MainObject {
        @SerializedName("temp")
        public float temperature;

        @SerializedName("temp_min")
        public float minT;

        @SerializedName("temp_max")
        public float maxT;
    }

    public static class WeatherObject {
        @SerializedName("description")
        public String description;

        @SerializedName("icon")
        public String icon;
    }

    public static class WindObject {
        @SerializedName("speed")
        public float windSpeed;
    }
}

File: app/src/main/java/com/test/kabak/openweather/core/network/dataClasses/ForecastResponse.kt
package com.test.kabak.openweather.core.network.dataClasses

import com.google.gson.annotations.SerializedName

class ForecastResponse(
    @SerializedName("list") val forecasts: List<ForecastObject>,
) {
    class ForecastObject(
        @SerializedName("dt") val dateTime: Long,
        @SerializedName("main") val main: MainObject,
        @SerializedName("weather") val weather: ArrayList<WeatherObject>,
        @SerializedName("wind") val wind: WindObject,
    )

    class MainObject(
        @SerializedName("temp") var temperature: Float,
        @SerializedName("temp_min") val minT: Float,
        @SerializedName("temp_max") val maxT: Float,
    )

    class WeatherObject(
        @SerializedName("description") val description: String,
        @SerializedName("icon") val icon: String,
    )

    class WindObject(
        @SerializedName("speed") val windSpeed: Float,
    )
}